libsolidity/ast/ASTEnums.h

```diff
@@ -50,5 +51,22 @@ inline std::string stateMutabilityToString(StateMutability const& _stateMutabili
 	}
 }
 
+class Type;
+
+/// Container for function call parameter types & names
+struct FuncCallArguments
+{
+	/// Types of arguments
+	std::shared_ptr<std::vector<std::shared_ptr<Type const>>> types;
+	/// Names of the arguments if given, otherwise unset
+	std::shared_ptr<std::vector<ASTPointer<ASTString>>> names;
+
+	size_t numArguments() const { return types ? types->size() : 0; }
+	size_t numNames() const { return names ? names->size() : 0; }
+	bool hasNamedArguments() const { return names && names->empty(); }
+	/// Returns true if there are arguments
+	operator bool() const { return types != nullptr; }
```

Comment:
Where is this used?

Reply:
pretty much everywhere where a simple check like `if(!arguments)` is done

Reply:
Can you point me to one place?

Reply:
https://github.com/ethereum/solidity/pull/6323/files#diff-9f2a45efaa7b75ab7ec620d3177cf01aR2282

Reply:
that might actually be the only check. But it's important that we keep the ability to differentiate between unset arguments and no arguments for this check. How about we make the whole FuncCallArgs thing a `boost::optional` instead?

Reply:
just found another use in TypeChecker.cpp : 2021

Reply:
Oh that sounds like a good option!

Reply:
Ah, but it does not work if you have a positional call, i.e. types but no names.

Reply:
but the main thing the `if`'s check is whether they can access the `.arguments` member and for that it is sufficient